python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -274,8 +272,12 @@ def __init__(
         self._pq_paths = [p.path for p in pq_ds.fragments]
         self._block_udf = _block_udf
         self._to_batches_kwargs = to_batch_kwargs
-        self._data_columns = data_columns
-        self._data_columns_rename_map = {}
+        # Store as projection_map (identity mapping if columns specified, None otherwise)
+        # Note: Empty list [] means no columns, None means all columns
+        if data_columns is None:
+            self._projection_map = None
+        else:
+            self._projection_map = {col: col for col in data_columns}
```

Comment:
Also out-of-scope for this PR, but I think it's implicit that you need to set this specific `_projection_map` attribute that's defined in an ancestor class, and it's not part of the `_DatasourceProjectionPushdownMixin` interface.